enumerables.rb

```diff
@@ -0,0 +1,154 @@
+module Enumerable
+    def my_each
+        return to_enum(:my_each) unless block_given?
+        is_range = self.kind_of?(Range) 
+        range = is_range ? self : 0...self.length
+        for i in range do
+            yield(is_range ? i : self[i])
+        end
+        return self unless !self.kind_of?(Array)
+        return self unless !self.kind_of?(Hash)
```

Comment:
Hello Samuel, we didn't know it should return self in any case. We were just fixing the test cases that the last TSE showed us were failling. It will be better if we have this testing file so we don't have to ask for a code review to see if everything is good.